pallets/runtime/tests/src/external_agents_test.rs

```diff
@@ -26,6 +31,15 @@ fn make_perms(pallet: &str) -> ExtrinsicPermissions {
     SubsetRestriction::elem(PalletPermissions::entire_pallet(pallet.into()))
 }
 
+fn add_become_agent(ticker: Ticker, from: User, to: User, group: AgentGroup) {
+    let data = AuthorizationData::BecomeAgent(ticker, group);
+    let sig = Signatory::Identity(to.did);
+    assert_ok!(Id::accept_authorization(
+        to.origin(),
+        Id::add_auth(from.did, sig, data, None)
+    ));
```

Comment:
What I was asking for is to dedup the creation of the auth itself, not the accepting bit, so that the `add` closure below gets simplified too.